extensions/GitExtension/FileExplorerGitIntegration/Models/GitLocalRepositoryProviderFactory.cs

```diff
@@ -34,8 +35,16 @@ GetLocalRepositoryResult ILocalRepositoryProvider.GetRepository(string rootPath)
         }
         catch (Exception ex)
         {
-            var log = Log.ForContext("SourceContext", nameof(GitLocalRepositoryProviderFactory));
-            log.Error("GitLocalRepositoryProviderFactory", "Failed to create GitLocalRepository", ex);
+            if (ex is LibGit2Sharp.RepositoryNotFoundException)
+            {
+                return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized("RepositoryNotFound"), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
+            }
+
+            if (ex.Message.Contains("not owned by current user") || ex.Message.Contains("detected dubious ownership in repository"))
+            {
+                return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized("RepositoryNotOwnedByCurrentUser"), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
+            }
+
             return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized(_errorResourceKey), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
         }

```

Comment:
I think it's a little easier to read and used more in the code currently to do this
```suggestion
        catch (LibGit2Sharp.RepositoryNotFoundException lg2sEx)
        {
            return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized("RepositoryNotFound"), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
        }
        catch (Exception ex)
        {
            if (ex.Message.Contains("not owned by current user") || ex.Message.Contains("detected dubious ownership in repository"))
            {
                return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized("RepositoryNotOwnedByCurrentUser"), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
            }

            return new GetLocalRepositoryResult(ex, _stringResource.GetLocalized(_errorResourceKey), $"Message: {ex.Message} and HRESULT: {ex.HResult}");
        }
```